### DISCUSSION TITLE
How to create my own simulation without modifying the existing example codes?

### DISCUSSION THREAD

**Discussion by LeSnow-Ye** [mfem/mfem#4605] [Q&A]
I'd like to use MFEM to do a simple simulation. But I'm new in C++, and I'm having touble including MFEM in another project. Is there a tutorial for using MFEM outside its source code or adding my own "miniapp" or .cpp file under the `examples` folder ?

**Comment by tzanio** [comment]
You can examine the makefiles in the miniapps directories, which are essentially external apps.

If you just want to add a new example-like file, say `my.cpp`, you can simply put it in the `examples/` directory and should be able to build it with `make my`.

**Reply by LeSnow-Ye** [reply]
I found [this repo](https://github.com/YacineBelaid/Compute-Cluster-Thermal-Diffusion-Simulation/) a few days ago. By referencing its `CMakeLists.txt` and using `cmake .. -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH=/path-to/mfem/`, I managed to get things done. Also, I'm using `cmake` to build MFEM following [HowTo/build-systems](https://mfem.org/howto/build-systems/). I haven't tried using `make` to build MFEM for external projects in this way.
